### DISCUSSION TITLE
How to submit an "old" version of an existing formula

### DISCUSSION THREAD

**Discussion by w43322** [Homebrew/discussions#4399] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.11
ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HEAD: 931327df1f30a3a5b1aa07657c0d1070e61988f9
Last commit: 10 days ago
Core tap JSON: 10 Apr 06:13 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_API_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles/api
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles
HOMEBREW_BREW_GIT_REMOTE: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/homebrew-core.git
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-x86_64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Some package `foo` introduced breaking changes in a new version, how can I add an older version of the formula in the `foo@x.x.x` format?

To be more specific, the current version of formula `v2ray` is 5.3.0, and it's a dependency of `qv2ray`. But package `qv2ray` only works with `v2ray 4.x or earlier`. When installing `qv2ray` using brew, the software doesn't work properly because it's not compatible with the dependency that's installed by brew.

I'm thinking about submitting a new formula `v2ray@4.45.2` and change the dependency of `qv2ray` from `v2ray` to `v2ray@4.45.2`. Are there any better alternatives? `qv2ray` is already abandoned so it's unrealistic to tell the devs to update their project...

**Comment by SMillerDev** [comment]
If a version meets https://docs.brew.sh/Versions#acceptable-versioned-formulae it can be added, but if the motivation is just to save abandonware, I don't expect that to be merged.
